css/css-grid/grid-model/grid-button-001.html

```diff
@@ -0,0 +1,26 @@
+<!DOCTYPE html>
+<title>Grid items work inside a button</title>
+<meta name="assert" content="When a button is set to display: grid, its children should flow into its grid cells">
+<link rel="author" title="Bryan Robinson" href="bryanlrobinson@gmail.com">
+<link rel="help" href="https://drafts.csswg.org/css-grid-1/#grid-containers">
+<style>
+  .grid { display: grid; grid-template-columns: 100px 200px; border: 2px solid red; box-sizing: border-box; /* for visual */}
+  span { border: 1px solid green; box-sizing: border-box; /* for visual */ }
+</style>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+
+<button class="grid"><span class="item1">item 1</span> <span class="item2">item 2</span></button>
```

Comment:
You can use `check-layout-th.js` and things like `data-expected-with` and even `data-offset-x` to be sure things are working as expected.

Reply:
Oh, that's handy! One question that I couldn't quite find in the source myself: Is there a way to do custom messages in the test using checkLayout() or is the default messaging good enough? Either way, it's super nice not to have to write all that myself (update incoming)

Reply:
Mmm, I'm not sure if there are anything like that. Usually the default message is clear enough.